README_cmake.md

```diff
@@ -1,106 +1,408 @@
+> cmake --build .\build
+```
+
+Otherwise, if you wish to create a Visual Studio based build system, you can
+configure with:
+
+```
+> cmake -G "Visual Studio 16 2019" -Thost=x64 -A x64 ^
+        -DCMAKE_TOOLCHAIN_FILE=D:\vcpkg\scripts\buildsystems\vcpkg.cmake ^
+        -S . -B build
+> cmake --build .\build --config Release -j %NUMBER_OF_PROCESSORS%
+```
+
+or, for 32-bit:
+
+```
+> cmake -G "Visual Studio 16 2019" -Thost=x64 -A Win32 ^
+        -DCMAKE_TOOLCHAIN_FILE=D:\vcpkg\scripts\buildsystems\vcpkg.cmake ^
+        -S . -B build
+> cmake --build .\build --config Release -j %NUMBER_OF_PROCESSORS%
+```
+
+In both cases, the `-Thost=x64` flag ensures that the correct host tools are
+used.
 
-This applies to WSL (Windows Subsystem for Linux) installations, too.
+### macOS and Linux
 
-## Using Halide in JIT mode
+The instructions here are straightforward. Assuming your environment is set up
+correctly, just run:
+
+```
+dev@host:~/Halide$ cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -S . -B build
+dev@host:~/Halide$ cmake --build ./build
+```
+
+## Installing
+
+Once built, Halide will need to be installed somewhere before using it in a
+separate project. On any platform, this means running the `cmake --install`
+command in one of two ways. For a single-configuration generator (like Ninja),
+run either:
+
+```
+dev@host:~/Halide$ cmake --install ./build --prefix /path/to/Halide-install
+> cmake --install .\build --prefix X:\path\to\Halide-install
+```
+
+For a multi-configuration generator (like Visual Studio) run:
+
+```
+dev@host:~/Halide$ cmake --install ./build --prefix /path/to/Halide-install --config Release
+> cmake --install .\build --prefix X:\path\to\Halide-install --config Release
+```
+
+Of course, make sure that you build the corresponding config before attempting
+to install it.
+
+## Build options
+
+Halide reads and understands several options that can configure the build. The
+following are the most consequential and control how Halide is actually
+compiled.
+
+| Option                       | Default               | Description                                                                                                      |
+| ---------------------------- | --------------------- | ---------------------------------------------------------------------------------------------------------------- |
+| `Halide_BUNDLE_LLVM`         | `OFF`                 | When building Halide as a static library, unpack the LLVM static libraries and add those objects to libHalide.a. |
+| `Halide_SHARED_LLVM`         | `OFF`                 | Link to the shared version of LLVM. Not available on Windows.                                                    |
+| `Halide_ENABLE_RTTI`         | _inherited from LLVM_ | Enable RTTI when building Halide. Recommended to be set to `ON`                                                  |
```

Comment:
Note that RTTI *must* be enabled in order to build/use the Python bindings for Halide.